### PR TITLE
Added touchable capacity

### PR DESCRIPTION
**Pull request by 3rdvision**
| - | - |
| --- | --- |
| Issue | Increase area of touch in key value component and add longPress functionality. Related to: |
| Decisions |   |
| Animated GIF |  |

### CODE REVIEW

**Comment by gcandal** [comment]
Didn't we lose the icon `onClick`? Besides that I'm fine with the changes, but we should probably have the 3: onClick, onLongClick and onItemClick.

**Comment by gcandal** [comment]
Also, any reason for this to be a draft?

**Comment by 3rdvision** [comment]
> Didn't we lose the icon `onClick`? Besides that I'm fine with the changes, but we should probably have the 3: onClick, onLongClick and onItemClick.

Adding `onButtonIconPress` for the icon click.

**Comment by 3rdvision** [comment]
> Also, any reason for this to be a draft?

Not really, I created both PRs on draft to first provide a better description with GIF and links. They are now ready to merge 😄 🚀

File: react/components/molecules/key-value/key-value.js [hunk 1]
```diff
@@ -31,13 +33,19 @@ export class KeyValue extends PureComponent {
             iconHeight: undefined,
             iconWidth: undefined,
             iconStrokeWidth: undefined,
-            iconOnPress: undefined
+            onPress: undefined,
+            onButtonIconPress: undefined,
+            onLongPress: undefined
```
Comment (gcandal) [inline react/components/molecules/key-value/key-value.js]: Use `() => {}` for coherence with other components.

File: react/components/molecules/key-value/key-value.js [hunk 2]
```diff
@@ -31,13 +33,19 @@ export class KeyValue extends PureComponent {
...
+            onPress: undefined,
+            onButtonIconPress: undefined,
+            onLongPress: undefined
         };
     }
 
     render() {
         return (
-            <View style={styles.keyValue}>
+            <TouchableOpacity
```
Comment (gcandal) [inline react/components/molecules/key-value/key-value.js]: This being touchable or not should be controlled by a `enablePress` prop. We don't want to make all key-values touchable, just some of them.

File: react/components/molecules/key-value/key-value.js [hunk 3]
```diff
@@ -18,7 +18,10 @@ export class KeyValue extends PureComponent {
             iconHeight: PropTypes.number,
             iconWidth: PropTypes.number,
             iconStrokeWidth: PropTypes.number,
-            iconOnPress: PropTypes.func
+            clickable: PropTypes.boolean,
```
Comment (gcandal) [inline react/components/molecules/key-value/key-value.js]: In mobile there are presses not clicks :D, please rename

Comment (3rdvision) [reply react/components/molecules/key-value/key-value.js]: Creating a PR to rename this as well on `ripe-robin-revamp`

**Pull request merged**